### ISSUE TITLE
pytorch - train/eval mode

### ISSUE THREAD

**Issue by junxnone** [junxnone/tio#647]
一些 Layers 训练时的froward和验证时的forward有计算上是不同的:
- BN
- Dropout

# Reference
- [ ] [train mode code](https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module.train)
- [ ] [BN Layers](https://pytorch.org/docs/stable/nn.html#normalization-layers)
- [ ] [Dropout Layers](https://pytorch.org/docs/stable/nn.html#dropout-layers)

# Brief

```
        self.training = mode
```
- train mode: True
- eval mode: False

## Dropout Layers
![image](https://user-images.githubusercontent.com/2216970/71708167-2e638680-2e2a-11ea-8984-cdc0b5ca1d20.png)

## BN Layers
![image](https://user-images.githubusercontent.com/2216970/71708122-e17fb000-2e29-11ea-903c-79af0dd896d6.png)
- Training： 每次输入都计算均值和方差，然后移动平均。
- Eval： 使用训练时的均值和方差，不再计算更新值

**Comment by junxnone** [comment]
#311
